File: 70-ClimbingStairs/70-ClimbingStairs.js
// Last updated: 4/9/2026, 11:12:45 AM
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if(n<2){
        return n;
    }

    let p1 = 1;
    let p2 = 2;

    for(let i = 3;i<=n;i++){
        let current = p1+p2;
        p1=p2;
        p2=current;
    }

    return p2;

};

File: 70-ClimbingStairs/70-ClimbingStairs.cpp
// Last updated: 4/9/2026, 11:12:42 AM
class Solution {
public:
    int climbStairs(int n) {
        if(n<2){
            return n;
        }

        int p1 = 1;
        int p2 = 2;

        for(int i = 3;i<=n;i++){
            int current = p1+p2;
            p1 = p2;
            p2 = current;
        }

        return p2;
    }
};